assets/sass/vendor/_mdc-checkbox.scss

```diff
@@ -31,6 +82,42 @@
 			vertical-align: middle;
 			width: 100%;
 			z-index: 1;
+
+			&:focus {
+				~ .mdc-checkbox__background::before {
+					opacity: 1;
+				}
+			}
+
+			&:enabled:not(:checked, :indeterminate),
+			&:checked:enabled {
+				~ .mdc-checkbox__background {
+					border-color: $c-black;
+				}
+			}
+
+			&:enabled:not(:checked, :indeterminate) {
+				~ .mdc-checkbox__background {
+					background-color: transparent;
+				}
+			}
+
+			&[disabled]:not(:checked, :indeterminate) {
+				~ .mdc-checkbox__background {
+					border-color: $c-interactive-on-disable-container;
+				}
+			}
+
+			&[disabled]:checked:not(:indeterminate) {
+				~ .mdc-checkbox__background {
+					background-color: $c-interactive-on-disable-container;
```

Comment:
Apologies, I realise I made a typo in my previous comment where I advised the correct colour token to use here. 

It should in fact be `$c-interactive-disable-container`.

```suggestion
					background-color: $c-interactive-disable-container;
```

<img width="1515" height="879" alt="image" src="https://github.com/user-attachments/assets/351f15c2-3d9d-4771-9956-f8cecf18d4fd" />